### PR TITLE
Javascript Library app

### PR DESCRIPTION
**Pull request by kirokoNjenga**
- Used Javascript prototypes to create functions
- Used Bootswatch for CSS(base on boostrap) for styling
- Used localStorage to keep data persistent

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  static deleteBook(el) {
+    if (el.classList.contains('delete')) {
+      el.parentElement.parentElement.remove();
+    }
+  }
+
+  static showAlert(message, className) {
+    const div = document.createElement('div');
+    div.className = `alert alert-${className}`;
```
Comment (bellom) [inline app.js]: Let's separate the DOM from application logic, you can create another file for DOM

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59623451/

File: index.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        <div class="form-group">
+          <label for="pages">Number of pages</label>
+          <input type="text" id="page" class="form-control" />
+        </div>
+        <input
+          type="submit"
+          value="Add Book"
+          class="btn btn-primary btn-block"
+        />
+      </form>
```
Comment (bellom) [inline index.html]: According to the project requirement, you're supposed to have a `button` that will bring up the form and users should be able to specify the status of the book either read or unread in the form.

image

https://www.theodinproject.com/courses/javascript/lessons/library

File: index.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      <table class="table table-striped mt-5">
+        <thead>
+          <tr>
+            <th>Title</th>
+            <th>Author</th>
+            <th>Number of pages</th>
+            <th></th>
+          </tr>
+        </thead>
+        <tbody id="book-list"></tbody>
```
Comment (bellom) [inline index.html]: From your app;


According to the project requirement, you're supposed to have a toggle feature, where users can change the book status.

image

https://www.theodinproject.com/courses/javascript/lessons/library

File: index.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+          <label for="title">Title</label>
+          <input type="text" id="title" class="form-control" />
+        </div>
+        <div class="form-group">
+          <label for="author">Author</label>
+          <input type="text" id="author" class="form-control" />
+        </div>
+        <div class="form-group">
+          <label for="pages">Number of pages</label>
+          <input type="text" id="page" class="form-control" />
```
Comment (bellom) [inline index.html]: Opps, you set the input page number to text, kindly validate this, users should not be able to enter non-digit data.
image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @Samkiroko  :clap: 

### Awesome Design, Good job so far!! :+1: 

**Quick One:** _**This project is one of the collaborative projects but you're working alone, may I ask why. kindly drop your answer in here for other TSE to see, thanks.**_

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by kirokoNjenga** [comment]
- I hand to work on almost a new UI but do all requested changes
- WHY am submitting the project alone: 
**I have been sick for the last two week and my partner is ahead of me, hope I will catch up**

**Review by Forison — CHANGES_REQUESTED** [review]
## Changes Required❌
# 🙋‍♂️ Samuel 
#### You have done a good job so far, however, some of the issues raised in the previous review still persist👨‍💼.
* [x]  There should be a new book button that brings up a form.
- As mentioned in the previous review, the should be an additional book button which on click brings up the book addition form, which is initially hidden until the button is clicked.
* [x] Kindly set an input validation on the number pages to make sure your number of pages only accepts a number. 
- Currently, "Number of pages" can be a string. Check the image below👇
check
- Kindly do the same for the read status, In the case of the status, a dropdown would be the best as the approach.

**[OPTIONAL]** 
* [ ] Th nature of the code in the app.js makes the reading of the code very hard variables, mixed with classes and functions. I would suggest you separate them and require/import when needed. This enhances code readability and future improvements.
* [ ] The UI can be improved and made more appealing, you can even add a picture of the book.
* [ ] Kindly add a description to the repository, this makes the repository more professional.
> Remember recruiters may ask you to submit some of the projects you have worked on during your job search. Having a good UI enhances and a description to all your repos is can be a plus.
### That is it, kindly make the changes and submit the project for a re-review🤗.
#### Happy coding😎
Addo Forison👨‍💻

**Comment by kirokoNjenga** [comment]
- @Forison I think you have checked the wrong PR.
- all the change that was requested I have done.
- Also, I have  UI as per the Readme file link: [LIVE DEMO](https://quizzical-yonath-9ac08b.netlify.app/)
- Added the validation

**Comment by Forison** [comment]
> * @Forison I think you have checked the wrong PR.
> * all the change that was requested I have done.
> * Also, I have  UI as per the Readme file link: [LIVE DEMO](https://quizzical-yonath-9ac08b.netlify.app/)
> * Added the validation

@Samkiroko I don't think I have the wrong branch, the other branches ie master and development are empty so they have nothing to show. I was trying to reach you on slack to see the source error but it seems you are not available currently. I will record and attached a video for the step to access the project as stated in the readme. My local machine's output is very different from the page you have in the live demo. I find this very strange. Kindly contact me on slack for further discussions on this.
[Please visit this link to see my findings](https://www.loom.com/share/f24ac464738d4acc9d329269a086ea9c)

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED 

Hi @Samkiroko this looks good to me, kudos for making changes please merge and move on :+1: 

**RAPHAEL NORIODE** :mask:

**Pull request merged**